src/OrchardCore/OrchardCore.Abstractions/Shell/Builders/Extensions/IAsyncOptions.cs

```diff
@@ -0,0 +1,9 @@
+namespace Microsoft.Extensions.Options;
+
+/// <summary>
+/// Marks a type of options intended to be registered as a singleton and configured asynchronously
+/// by an <see cref="IAsyncConfigureOptions{TOptions}"/> just after a tenant container is created.
+/// </summary>
+public interface IAsyncOptions
```

Comment:
Can't we use the built in IOptions?

Reply:
Sorry ;) Would need async service resolutions, async constructor.

See https://github.com/OrchardCMS/OrchardCore/pull/14759/files#r1405670950